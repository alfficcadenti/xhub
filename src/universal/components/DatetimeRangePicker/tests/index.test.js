import React from 'react';
import {DatetimeRangePicker} from '../index.js';
import {render, fireEvent, act, screen, within} from '@testing-library/react';
import '@testing-library/jest-dom';
import moment from 'moment';


describe('DatetimeRangePicker component testing', () => {
    const dates = {end: new Date('2022-02-01'), start: new Date('2022-01-01')};
    const finalDates = {end: new Date('2022-02-28'), start: new Date('2021-12-26')};
    it('renders successfully 2 inputs (start and end)', async () => {
        const wrapper = render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={() => {}} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.getByLabelText('Start')).toBeInTheDocument();
        expect(wrapper.getByLabelText('End')).toBeInTheDocument();
    });

    it('assign the active class when an input node is clicked by the user', async () => {
        const onChange = jest.fn();
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(onChange).toHaveBeenCalledTimes(1);
        expect(inputs[0]).not.toHaveClass('active');
        fireEvent.click(inputs[0]);
        expect(inputs[0]).toHaveClass('active');
        fireEvent.click(screen.getAllByText('26')[0]);
        expect(onChange).toHaveBeenCalledTimes(2);
    });

    it('assign the active class when an input node is clicked by the user', async () => {
        const onChange = jest.fn();
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(inputs[0]).not.toHaveClass('active');
        fireEvent.click(inputs[0]);
        expect(inputs[0]).toHaveClass('active');
        fireEvent.click(screen.getAllByText('26')[0]);
        fireEvent.click(screen.getAllByText('28')[0]);
        expect(inputs[0]).not.toHaveClass('active');
        expect(onChange).toHaveBeenLastCalledWith(finalDates);
    });

    it('close the calendar when click outside', async () => {
        const onChange = jest.fn();
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(inputs[0]).not.toHaveClass('active');
        fireEvent.click(inputs[0]);
        expect(inputs[0]).toHaveClass('active');
        fireEvent.click(screen.getAllByText('26')[0]);
        expect(inputs[1]).toHaveClass('active');
        fireEvent.mouseDown(document);
        expect(inputs[1]).not.toHaveClass('active');
    });

    it('assign a start date when user input in textboxes', async () => {
        const onChange = jest.fn();
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(onChange).toHaveBeenCalledTimes(1);
        fireEvent.click(inputs[0]);
        fireEvent.change(screen.queryAllByRole(/textbox/)[0], {target: {value: '2021-12-26'}});
        expect(onChange).toHaveBeenCalledTimes(2);
        expect(screen.queryByLabelText('Start')).toHaveFocus();
        fireEvent.change(screen.queryAllByRole(/textbox/)[1], {target: {value: '2022-02-28'}});
        expect(screen.queryByLabelText('End')).toHaveFocus();
        expect(onChange).toHaveBeenCalledTimes(3);
        expect(onChange).toHaveBeenLastCalledWith(finalDates);
        fireEvent.keyUp(screen.queryAllByRole(/textbox/)[1], {key: 'Enter', code: 13, charCode: 13});
        expect(screen.queryByLabelText('Start')).not.toHaveFocus();
        expect(screen.queryByLabelText('End')).not.toHaveFocus();
    });

    it('assigns a start date 1 day before the end date if the user select an end date which is earlier than the current start date', async () => {
        const onChange = jest.fn();
        const finalValidDates = {end: new Date('2022-02-28'), start: new Date('2022-02-27')};
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(onChange).toHaveBeenCalledTimes(1);
        fireEvent.click(inputs[0]);
        fireEvent.change(screen.queryAllByRole(/textbox/)[0], {target: {value: '2022-03-26'}});
        expect(onChange).toHaveBeenCalledTimes(2);
        fireEvent.change(screen.queryAllByRole(/textbox/)[1], {target: {value: '2022-02-28'}});
        expect(onChange).toHaveBeenCalledTimes(3);
        expect(onChange).toHaveBeenLastCalledWith(finalValidDates);
    });

    it('assigns a start date 1 hour before the end date if the user select an end date which is earlier than the current start date and the time format is available', async () => {
        const onChange = jest.fn();
        const finalValidDates = {end: new Date('2022-02-28 04:32'), start: new Date('2022-02-28 03:32')};
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} showTimePicker/>);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        expect(onChange).toHaveBeenCalledTimes(1);
        fireEvent.click(inputs[0]);
        fireEvent.change(screen.queryAllByRole(/textbox/)[0], {target: {value: '2022-03-26 05:40'}});
        expect(onChange).toHaveBeenCalledTimes(2);
        fireEvent.change(screen.queryAllByRole(/textbox/)[1], {target: {value: '2022-02-28 04:32'}});
        expect(onChange).toHaveBeenCalledTimes(3);
        expect(onChange).toHaveBeenLastCalledWith(finalValidDates);
    });

    it('assigns a end date 1 hour after the start date if the user select a start date which is later than the current end date and the time format is available', async () => {
        const onChange = jest.fn();
        const finalValidDates = {end: new Date('2022-05-28 05:32'), start: new Date('2022-05-28 04:32')};
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} showTimePicker/>);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        fireEvent.click(inputs[0]);
        fireEvent.change(screen.queryAllByRole(/textbox/)[0], {target: {value: '2022-05-28 04:32'}});
        expect(onChange).toHaveBeenLastCalledWith(finalValidDates);
    });

    it('assigns new end date if user click on end date input textbox before selecting a date from the calendar', async () => {
        const onChange = jest.fn();
        const finalValidDates = {end: new Date('2022-02-26'), start: new Date('2022-01-01')};
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let inputs = screen.queryAllByRole(/textbox/);
        fireEvent.click(inputs[1]);
        fireEvent.click(screen.getAllByText('26')[0]);
        expect(onChange).toHaveBeenLastCalledWith(finalValidDates);
    });

    it('updates start and end date if user select a preset timeframe', async () => {
        const onChange = jest.fn();
        const lastWeek = moment().subtract(1, 'week').startOf('day').toDate();
        const today = moment().endOf('day').toDate();
        const finalValidDates = {end: today, start: lastWeek};
        await act(async () => {
            render(<DatetimeRangePicker startDate={dates.start} endDate={dates.end} onChange={onChange} />);
        });
        let preset = screen.queryByRole('dialog');
        fireEvent.click(preset);
        fireEvent.click(within(preset).getAllByRole('presentation')[0]);
        expect(onChange).toHaveBeenLastCalledWith(finalValidDates, 'Last 7 Days');
    });
});
