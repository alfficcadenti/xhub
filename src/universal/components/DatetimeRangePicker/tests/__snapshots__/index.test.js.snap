// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeRangePicker component testing renders successfully 2 inputs (start and end) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="datetime-range-picker"
      >
        <div
          class="rdt-container "
          role="button"
        >
          <div
            class="rdt-input-container"
          >
            <div
              class="rdt-start-container"
            >
              <label
                class="rdt-start-label"
                for="datepicker-start-date"
              >
                Start
              </label>
              <div
                class="rdt"
              >
                <input
                  class="form-control short "
                  id="datepicker-start-date"
                  type="text"
                  value="01/01/2022"
                />
              </div>
            </div>
            <div
              class="rdt-end-container"
            >
              <label
                class="rdt-end-label"
                for="datepicker-end-date"
              >
                End
              </label>
              <div
                class="rdt"
              >
                <input
                  class="form-control short "
                  id="datepicker-end-date"
                  type="text"
                  value="02/01/2022"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Dropdown rdt-preset-dropdown"
          role="presentation"
          tabindex="-1"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="btn btn-default Dropdown__toggle border-border-color border--contrast"
            id="rdt-preset-dropdown"
            type="button"
          >
            <span
              class="btn__label"
            >
              Custom
            </span>
            <span
              class="SVGIcon SVGIcon--12px Dropdown__toggle-icon"
            >
              <svg
                data-id="SVG_CHEVRON_DOWN__12"
                focusable="false"
                height="12"
                viewBox="0 0 12 12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 4L6 9 1 4"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </button>
          <ul
            aria-labelledby="rdt-preset-dropdown"
            class="Dropdown__menu"
            id="rdt-preset-dropdown--container"
            role="dialog"
          >
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last 7 Days
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last 14 Days
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last 30 Days
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last 60 Days
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last 90 Days
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Last Year
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                All Months
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                May 2022
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Apr 2022
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Mar 2022
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Feb 2022
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jan 2022
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Dec 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Nov 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Oct 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Sep 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Aug 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jul 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jun 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                May 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Apr 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Mar 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Feb 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jan 2021
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Dec 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Nov 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Oct 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Sep 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Aug 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jul 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jun 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                May 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Apr 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Mar 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Feb 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jan 2020
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Dec 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Nov 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Oct 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Sep 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Aug 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jul 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jun 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                May 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Apr 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Mar 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Feb 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jan 2019
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Dec 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Nov 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Oct 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Sep 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Aug 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jul 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jun 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                May 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Apr 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Mar 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Feb 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Jan 2018
              </a>
            </li>
            <li
              role="presentation"
            >
              <a
                data-bypass="false"
                href="#"
              >
                 
                Dec 2017
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="datetime-range-picker"
    >
      <div
        class="rdt-container "
        role="button"
      >
        <div
          class="rdt-input-container"
        >
          <div
            class="rdt-start-container"
          >
            <label
              class="rdt-start-label"
              for="datepicker-start-date"
            >
              Start
            </label>
            <div
              class="rdt"
            >
              <input
                class="form-control short "
                id="datepicker-start-date"
                type="text"
                value="01/01/2022"
              />
            </div>
          </div>
          <div
            class="rdt-end-container"
          >
            <label
              class="rdt-end-label"
              for="datepicker-end-date"
            >
              End
            </label>
            <div
              class="rdt"
            >
              <input
                class="form-control short "
                id="datepicker-end-date"
                type="text"
                value="02/01/2022"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="Dropdown rdt-preset-dropdown"
        role="presentation"
        tabindex="-1"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="btn btn-default Dropdown__toggle border-border-color border--contrast"
          id="rdt-preset-dropdown"
          type="button"
        >
          <span
            class="btn__label"
          >
            Custom
          </span>
          <span
            class="SVGIcon SVGIcon--12px Dropdown__toggle-icon"
          >
            <svg
              data-id="SVG_CHEVRON_DOWN__12"
              focusable="false"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 4L6 9 1 4"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
        <ul
          aria-labelledby="rdt-preset-dropdown"
          class="Dropdown__menu"
          id="rdt-preset-dropdown--container"
          role="dialog"
        >
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last 7 Days
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last 14 Days
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last 30 Days
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last 60 Days
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last 90 Days
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Last Year
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              All Months
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              May 2022
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Apr 2022
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Mar 2022
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Feb 2022
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jan 2022
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Dec 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Nov 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Oct 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Sep 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Aug 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jul 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jun 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              May 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Apr 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Mar 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Feb 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jan 2021
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Dec 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Nov 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Oct 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Sep 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Aug 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jul 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jun 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              May 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Apr 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Mar 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Feb 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jan 2020
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Dec 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Nov 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Oct 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Sep 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Aug 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jul 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jun 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              May 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Apr 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Mar 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Feb 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jan 2019
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Dec 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Nov 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Oct 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Sep 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Aug 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jul 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jun 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              May 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Apr 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Mar 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Feb 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Jan 2018
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              data-bypass="false"
              href="#"
            >
               
              Dec 2017
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
